cmd/register/files.go

```diff
@@ -0,0 +1,179 @@
+		_, err := cmdCtx.AdminClient().CreateTask(ctx, &admin.TaskCreateRequest{
+			Id: &core.Identifier{
+				ResourceType: core.ResourceType_TASK,
+				Project:      config.GetConfig().Project,
+				Domain:       config.GetConfig().Domain,
+				Name:         name,
+				Version:      GetConfig().version,
+			},
+			Spec: message.(*admin.TaskSpec),
+		})
+		return err
+	default:
+		return errors.New(fmt.Sprintf("Failed registering unknown entity  %v type as part of %v file", message, name))
+	}
+}
+
+func hydrateNode(node *core.Node) {
+	targetNode := node.Target
+	switch targetNode.(type) {
+	case *core.Node_TaskNode:
+		taskNodeWrapper := targetNode.(*core.Node_TaskNode)
+		taskNodeReference := taskNodeWrapper.TaskNode.Reference.(*core.TaskNode_ReferenceId)
+		hydrateIdentifier(taskNodeReference.ReferenceId)
+	case *core.Node_WorkflowNode:
+		workflowNodeWrapper := targetNode.(*core.Node_WorkflowNode)
+		switch workflowNodeWrapper.WorkflowNode.Reference.(type) {
+		case *core.WorkflowNode_SubWorkflowRef:
+			subWorkflowNodeReference := workflowNodeWrapper.WorkflowNode.Reference.(*core.WorkflowNode_SubWorkflowRef)
+			hydrateIdentifier(subWorkflowNodeReference.SubWorkflowRef)
+		case *core.WorkflowNode_LaunchplanRef:
+			launchPlanNodeReference := workflowNodeWrapper.WorkflowNode.Reference.(*core.WorkflowNode_LaunchplanRef)
+			hydrateIdentifier(launchPlanNodeReference.LaunchplanRef)
+		}
+	}
+}
+
+func hydrateIdentifier(identifier *core.Identifier) {
+	identifier.Project = config.GetConfig().Project
+	identifier.Domain = config.GetConfig().Domain
+	identifier.Version = GetConfig().version
+}
+
+func hydrateSpec(message proto.Message) {
+	switch message.(type) {
+	case *admin.LaunchPlanSpec:
+		launchSpec := message.(*admin.LaunchPlanSpec)
+		hydrateIdentifier(launchSpec.WorkflowId)
+	case *admin.WorkflowSpec:
+		workflowSpec := message.(*admin.WorkflowSpec)
+		for _, Noderef := range workflowSpec.Template.Nodes {
+			hydrateNode(Noderef)
+		}
+		hydrateIdentifier(workflowSpec.Template.Id)
+	case *admin.TaskSpec:
+		taskSpec := message.(*admin.TaskSpec)
+		hydrateIdentifier(taskSpec.Template.Id)
+	}
+}
+
+func registerFromFilesFunc(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
+	files := args
+	sort.Strings(files)
+	logger.Infof(ctx, "Parsing files... Total(%v)", len(files))
+	logger.Infof(ctx, "Params version %v", GetConfig().version)
+	for _, absFilePath := range files {
+		if strings.Contains(absFilePath, identifierFileSuffix) {
+			continue
+		}
+		logger.Infof(ctx, "Parsing  %v", absFilePath)
+		fileContents, err := ioutil.ReadFile(absFilePath)
+		if err != nil {
+			logger.Errorf(ctx, "Error reading file %v  due to : %v. Skipping", absFilePath, err)
+			continue
+		}
+		spec, err := unMarshalContents(ctx, fileContents, absFilePath)
+		if err != nil {
+			logger.Errorf(ctx, "Error unmarshalling Skipping file %v due to : %v", absFilePath, err)
+			continue
+		}
+		name := getEntityNameFromPath(absFilePath)
```

Comment:
put a TODO here?

Reply:
Removed this functionality now that we have flytekit 0.16.0b2 release.